app/src/main/java/edu/wisc/ece/pinpoint/pages/map/MapFragment.java

```diff
@@ -173,6 +176,23 @@ private void styleMap() {
         }
     }
 
+    private void setMarkerWindowFunction(){
+        if (getActivity() != null){
+            map.setInfoWindowAdapter(new InfoAdapter(requireContext()));
+            map.setOnMarkerClickListener(mark -> {
+
+                //call once to force an image load
+                mark.showInfoWindow();
+
+                //call a second time to set the image (should actually be loaded by this time)
+                final Handler handler = new Handler(Looper.getMainLooper());
+                handler.postDelayed(() -> mark.showInfoWindow(), MARKER_IMAGE_LOAD_TIME);
```

Comment:
Is it possible to make this a while loop with exponentially increasing wait times? I could see a client with slow internet not being able to load the image within 200ms.